growthbook/growthbook
# Duplicate features

### Features and Changes

This PR adds a 'Duplicate' workflow to the features page.

- Adds a `MoreMenu` to rows on the features page, with a 'Duplicate' option.
- When duplicating, we open the `FeatureModal` with a `featureToDuplicate` prop supplied. This hides most of the form and only allows for editing the feature id (which must be unique).
- Duplicated features are created with  the same value type, tags, rules, and default value. However, id and description are not copied. 
  - If a user duplicates a feature with environments that they don't have publish features for, the duplicated feature's enabled state for relevant environments will be forced to `false` 
- Removes `id` prop on `MoreMenu` since it is now internally maintained by the component (uniquely generated id via `uniqid`)


- Closes #771 

### Testing

- Create a new feature with rules, overrides, tags, etc.
- See feature in list, click the more menu and select 'Duplicate'
- Duplicate feature with new id
- Validate the new, duplicated feature matches original wrt. rules, overrides, tags, etc.

### Screenshots

<img width="351" alt="image" src="https://user-images.githubusercontent.com/2374625/206479937-3dfa9ac6-531c-43b9-bb3a-03d3a79e396d.png">

<img width="851" alt="image" src="https://user-images.githubusercontent.com/2374625/206480010-4a09d1a0-6e7e-437f-be44-dcebed2541c0.png">

original feature vs. duped feature

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/2374625/206480352-a5220041-12eb-4500-ab82-742a077c9d07.png">


<img width="1624" alt="image" src="https://user-images.githubusercontent.com/2374625/206480237-fa275db1-c661-44fa-8adb-aff9db3520bc.png">
